ReSharper.FSharp/src/FSharp.Psi.Features/src/Intentions/ToInterpolatedString.fs

```diff
@@ -0,0 +1,155 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Intentions
+
+open System.Text
+open JetBrains.DocumentModel
+open JetBrains.ReSharper.Feature.Services.ContextActions
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Util
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Feature.Services.Daemon
+
+type private StringManipulation =
+    | InsertInterpolation of specifier: string * expr: IFSharpExpression
+    | EscapeBrace of braceChar: char
+
+[<ContextAction(Name = "ToInterpolatedString", Group = "F#",
+                Description = "Convert an printf-style format string with an interpolated string")>]
+type ToInterpolatedStringAction(dataProvider: FSharpContextActionDataProvider) =
+    inherit FSharpContextActionBase(dataProvider)
+
+    let [<Literal>] opName = "ToInterpolatedStringAction"
+
+    override x.Text = "To interpolated string"
+
+    override x.IsAvailable _ =
+        let prefixAppExpr = dataProvider.GetSelectedElement<IPrefixAppExpr>()
```

Comment:
I think it should only be available when caret is at `sprintf`-like function identifier or at the corresponding string literal. Making it available in more places (e.g. at the `sprintf` argument) will likely interfere with other actions in future. Take a look at the existing actions for examples.

Reply:
I need to add the availability tests, but at the moment it's only available on the `sprintf` or the format string itself - not anywhere else.

Reply:
Just pushed some availability tests